### PR TITLE
Add appendBytes method to appender

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
Add `appendBytes` method to `appender`.

### CODE REVIEW

File: src/test/java/org/duckdb/TestDuckDBJDBC.java [hunk 1]
```diff
@@ -2635,6 +2635,27 @@ public static void test_appender_null_varchar() throws Exception {
...
 
+    public static void test_appender_bytes() throws Exception {
+        DuckDBConnection conn = DriverManager.getConnection(JDBC_URL).unwrap(DuckDBConnection.class);
+        Statement stmt = conn.createStatement();
+
+        stmt.execute("CREATE TABLE data (a BLOB)");
+        DuckDBAppender appender = conn.createAppender(DuckDBConnection.DEFAULT_SCHEMA, "data");
+
+        appender.beginRow();
+        appender.appendBytes("test".getBytes(StandardCharsets.UTF_8));
```
Comment (Mause) [inline src/test/java/org/duckdb/TestDuckDBJDBC.java:2646]: Can you add a test with binary data, invalid utf-8, and null bytes?

**Review by Mause — CHANGES_REQUESTED** [review]

**Comment by staticlibs** [comment]
Closing this as superseded by #295.
